app/serializers/server_room_serializer.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class ServerRoomSerializer < RoomSerializer
+  attributes :owner, :status, :participants
+
+  def owner
+    object.user.name
+  end
+
+  def status
```

Comment:
Instead of iterating over the `active_server_rooms` each time for each room why don't we do something similar to the shared logic that @farhatahmad introduced.
Where we set `participants` and `status` as room instance attributes that can be read `attr_reader` and on the `#index` action level we iterate over our filtered list of rooms and simply check wither there's a matching meeting on the server and set those attributes accordingly.
It feels more natural that serializers be concerned only by representing the data and that actions encapsulate the logic behind providing serialalizers with the right data to represent. 
Also this has some **minor** performance implication.

Reply:
Now that I think about it, that might make more sense. Set `attr_reader` in the controller before being passed into the serializer (similar to how shared rooms is done)

Reply:
Putting the logic in the serializer definitely feels a bit weird.
I'm not familiar with attr_accessor - and not sure of the implications - but I'm reading on it.